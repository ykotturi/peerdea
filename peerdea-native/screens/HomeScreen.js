import * as WebBrowser from 'expo-web-browser';
import React, {Component} from 'react';
import {
  Button,
  Image,
  Platform,
  ScrollView,
  Text,
  TouchableOpacity,
  TouchableHighlight,
  View,
  Dimensions,
} from 'react-native';
import styles from '../components/style';
import statusCheck from "../components/StatusCheck.js";


export default class HomeScreen extends React.Component {
  static navigationOptions = {
    header: null
  };

  componentDidMount() {
    this.focusListener = this.props.navigation.addListener(
      'focus',
      async () => {
        var status = await statusCheck();
        if (status == "down"){
          this.props.navigation.navigate("Maintenance");
          return;
        }
      },
    );
  }

  render() {
    const {navigate} = this.props.navigation;
    return (
      <View style={styles.container}>

          <Image
            style={{width: 300, height: 50}}
            source={require('../assets/images/peerdea-logo-draft.png')}
            accessible={true}
            accessibilityLabel="Peerdea app logo"
          />

          <Text style={styles.getStartedText}>Welcome to Peerdea</Text>

          <TouchableHighlight
           style={styles.buttonSmallLeft}
           onPress={() => navigate('CreateUser')}
          >
            <Text style={styles.buttonText}>
              Sign up
            </Text>
          </TouchableHighlight>

          <View style={{padding: 5}} />

          <TouchableHighlight
           style={styles.buttonSmallLeft}
           onPress={() => navigate('UserLogin')}
          >
            <Text style={styles.buttonText}>
              Log in
            </Text>
          </TouchableHighlight>

          <View style={{padding: 5}} />

          <TouchableHighlight
           underlayColor="#CECECE"
           onPress={() => navigate('Onboarding')}
          >
            <Text style={{fontSize: 18, color: "#777777", textDecorationLine: "underline"}}>
              Learn more about Peerdea!
            </Text>
          </TouchableHighlight>

      </View>
    );
  }
}
