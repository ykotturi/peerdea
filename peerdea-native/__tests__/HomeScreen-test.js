import React from 'react';
import renderer from 'react-test-renderer';

import HomeScreen from '../screens/HomeScreen';

describe('Home Screen', () => {
  jest.useFakeTimers();

  it(`renders the home screen`, () => {
    const mockNavigation = {
      navigate: jest.fn(),
      addListener: jest.fn().mockImplementation((event, callback) => {
        callback();
        //returning value for `navigationSubscription`
        return { remove: jest.fn() }
      }),
    };

    const tree = renderer.create(
      <HomeScreen navigation={mockNavigation} />);
    expect(tree.toJSON()).toMatchSnapshot();

    var textList = tree.root.findAllByType('Text')
    
    textList.forEach((el) => {
      if (el.props.children === 'Sign up') {
        el.parent.parent.props.onClick()
        expect(mockNavigation.navigate).toHaveBeenCalledWith('CreateUser')
      }
      else if (el.props.children === 'Log in') {
        el.parent.parent.props.onClick()
        expect(mockNavigation.navigate).toHaveBeenCalledWith('UserLogin')
      }
      else if (el.props.children === 'Learn more about Peerdea!') {
        el.parent.parent.props.onClick()
        expect(mockNavigation.navigate).toHaveBeenCalledWith('Onboarding')
      }
    })
  });
});
