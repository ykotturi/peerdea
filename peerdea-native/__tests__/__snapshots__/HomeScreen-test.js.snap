// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen renders the home screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingBottom": 30,
      "paddingTop": 18,
    }
  }
>
  <Image
    accessibilityLabel="Peerdea app logo"
    accessible={true}
    source={1}
    style={
      Object {
        "height": 50,
        "width": 300,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "rgba(96,100,109, 1)",
        "fontSize": 17,
        "lineHeight": 20,
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to Peerdea
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0076ff",
        "borderColor": "#0076ff",
        "borderRadius": 9,
        "borderWidth": 1,
        "elevation": 3,
        "height": 50,
        "justifyContent": "center",
        "width": "45%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "center",
          "color": "#ffffff",
          "fontSize": 20,
          "justifyContent": "center",
          "paddingHorizontal": "8%",
        }
      }
    >
      Sign up
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 5,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0076ff",
        "borderColor": "#0076ff",
        "borderRadius": 9,
        "borderWidth": 1,
        "elevation": 3,
        "height": 50,
        "justifyContent": "center",
        "width": "45%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "center",
          "color": "#ffffff",
          "fontSize": 20,
          "justifyContent": "center",
          "paddingHorizontal": "8%",
        }
      }
    >
      Log in
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 5,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        Object {
          "color": "#777777",
          "fontSize": 18,
          "textDecorationLine": "underline",
        }
      }
    >
      Learn more about Peerdea!
    </Text>
  </View>
</View>
`;
